core/workspace.js

```diff
@@ -301,36 +301,42 @@ Blockly.Workspace.prototype.getVariableUses = function(name) {
  * @param {string} name Name of variable to delete.
  */
 Blockly.Workspace.prototype.deleteVariable = function(name) {
+  var workspace = this;
   var variableIndex = this.variableIndexOf(name);
   if (variableIndex != -1) {
     var uses = this.getVariableUses(name);
+    function doDeletion() {
+      Blockly.Events.setGroup(true);
+      for (var i = 0; i < uses.length; i++) {
+        uses[i].dispose(true, false);
+      }
+      Blockly.Events.setGroup(false);
+      workspace.variableList.splice(variableIndex, 1);
+    }
     if (uses.length > 1) {
+      // Confirm the deletion of multiple blocks
```

Comment:
No, that's the confirm call on line (new) 329.  This is // Don't delete the variable if it's used in the definition of a procedure.